### PR TITLE
Enable $compute allowed settings

### PR DESCRIPTION
**Pull request by xuzhg**
issue #730

Enable to validate $compute if AllowedQueryOptions has correct setting.

### CODE REVIEW

File: src/Microsoft.AspNetCore.OData/Query/Validator/ODataQueryValidator.cs [hunk 1]
```diff
@@ -36,18 +36,13 @@ public virtual void Validate(ODataQueryOptions options, ODataValidationSettings
             // Validate each query options
             if (options.Compute != null)
             {
-                if (options.Compute.ComputeClause != null)
-                {
-                    ValidateQueryOptionAllowed(AllowedQueryOptions.Compute, validationSettings.AllowedQueryOptions);
-                }
+                ValidateQueryOptionAllowed(AllowedQueryOptions.Compute, validationSettings.AllowedQueryOptions);
+                options.Compute.Validate(validationSettings);
```
Comment (dnperfors) [inline src/Microsoft.AspNetCore.OData/Query/Validator/ODataQueryValidator.cs:40]: This change leads to not parsing the ComputeClause during this validation. Is that desirable?

Comment (xuzhg) [reply src/Microsoft.AspNetCore.OData/Query/Validator/ODataQueryValidator.cs:40]: Actually,  in ODataQueryValidator, we validate whether the query option is allowed or not. 
In each Query validator, for example FilterQueryValidator, ComputeQueryValidator, we valid the content. 

For $compute clause, what expect error message do you want to get if the $compute clause is not valid?

Comment (dnperfors) [reply src/Microsoft.AspNetCore.OData/Query/Validator/ODataQueryValidator.cs:40]: If that is the way, other validators work as well, than that is fine with me.
If the ComputeQueryValidator will parse the string and returns error message when the expression is incorrect, than that would be good enough for me, just like the filter errors.

**Review by KenitoInc — APPROVED** [review]
LGTM

File: src/Microsoft.AspNetCore.OData/Query/Validator/ODataQueryValidator.cs (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.AspNetCore.OData/Query/Validator/ODataQueryValidator.cs:40]: I will merge this PR and Let me try adding the expression validation in another PR.

**Pull request merged**